include/zenoh-pico/utils/logging.h

```diff
@@ -17,55 +17,57 @@
 
 #include <stdio.h>
 
+// Logging values
+#define _Z_LOG_LVL_ERROR 1
+#define _Z_LOG_LVL_INFO 2
+#define _Z_LOG_LVL_DEBUG 3
+
+// Timestamp function
 static inline void __z_print_timestamp(void) {
     char ret[64];
     printf("[%s ", z_time_now_as_str(ret, sizeof(ret)));
 }
 
+// Logging macros
 #define _Z_LOG_PREFIX(prefix) \
     __z_print_timestamp();    \
     printf(#prefix " ::%s] ", __func__);
 
-#if (ZENOH_DEBUG == 3)
-#define _Z_DEBUG(x, ...)  \
-    _Z_LOG_PREFIX(DEBUG); \
-    printf(x, ##__VA_ARGS__);
-#define _Z_DEBUG_CONTINUE(x, ...) printf(x, ##__VA_ARGS__);
-#define _Z_INFO(x, ...)  \
-    _Z_LOG_PREFIX(INFO); \
-    printf(x, ##__VA_ARGS__);
-#define _Z_INFO_CONTINUE(x, ...) printf(x, ##__VA_ARGS__);
-#define _Z_ERROR(x, ...)  \
-    _Z_LOG_PREFIX(ERROR); \
-    printf(x, ##__VA_ARGS__);
-#define _Z_ERROR_CONTINUE(x, ...) printf(x, ##__VA_ARGS__);
+// Ignore print only if log deactivated and build is release
+#if ZENOH_DEBUG == 0 && !defined(Z_BUILD_DEBUG)
+
+#define _Z_DEBUG(...) (void)(0)
+#define _Z_INFO(...) (void)(0)
+#define _Z_ERROR(...) (void)(0)
+
+#else  // ZENOH_DEBUG != 0 || defined(Z_BUILD_DEBUG)
 
-#elif (ZENOH_DEBUG == 2)
-#define _Z_DEBUG(x, ...) (void)(0)
-#define _Z_DEBUG_CONTINUE(x, ...) (void)(0);
-#define _Z_INFO(x, ...)  \
-    _Z_LOG_PREFIX(INFO); \
-    printf(x, ##__VA_ARGS__);
-#define _Z_INFO_CONTINUE(x, ...) printf(x, ##__VA_ARGS__);
-#define _Z_ERROR(x, ...)  \
-    _Z_LOG_PREFIX(ERROR); \
-    printf(x, ##__VA_ARGS__);
-#define _Z_ERROR_CONTINUE(x, ...) printf(x, ##__VA_ARGS__);
+#define _Z_DEBUG(...)                          \
+    do {                                       \
+        if (ZENOH_DEBUG >= _Z_LOG_LVL_DEBUG) { \
```

Comment:
Why to do this check at runtime, instead of at compile time as before?

Reply:
Actually it transitions from a pre-compiler thing to a compile time check, as the compiler realises this values won't change for the whole execution of the program, a bit like a C++ `constexpr`. But just to be extra sure I added the `(void)(0)` in release mode and no debug, in case some exotic compiler emits run time checks.

The problem with pre compiler stuff is that it doesn't check if the code is valid while the compiler do. So let's say in my log statement I print a variable, that is then changed (type or name), it will go unnoticed until the compiler sees it, so if I only compile without Debug it won't see it. Now it will always see this kind of issues.

Reply:
I am not sure if it is already in place, but having compilation tests for both debug and release seems a simpler option.